src/string_util.cc

```diff
@@ -92,21 +91,24 @@ std::string ExponentToPrefix(int64_t exponent, bool iec) {
   const int64_t index = (exponent > 0 ? exponent - 1 : -exponent - 1);
   if (index >= kUnitsSize) return "";
 
-  const char* array =
+  const std::string* array =
       (exponent > 0 ? (iec ? kBigIECUnits : kBigSIUnits) : kSmallSIUnits);
   if (iec) {
-    return array[index] + std::string("i");
+    return array[index] + "i";
   }
-  return std::string(1, array[index]);
+  return array[index];
 }
 
 std::string ToBinaryStringFullySpecified(double value, double threshold,
-                                         int precision, double one_k = 1024.0) {
+                                         int precision, benchmark::Counter::OneK one_k = benchmark::Counter::kIs1024) {
   std::string mantissa;
   int64_t exponent;
   ToExponentAndMantissa(value, threshold, precision, one_k, &mantissa,
                         &exponent);
-  return mantissa + ExponentToPrefix(exponent, false);
+  // Determine whether to use IEC units
+  bool use_iec = (one_k == benchmark::Counter::kIs1024);
+
+  return mantissa + ExponentToPrefix(exponent, use_iec);
```

Comment:
why not pass `one_k` directly to `ExponentToPrefix`?

Reply:
I have observed this style of passing in a file while working on the documentation.
Next time, I will be careful (my poor coding practices 🙂)